deployments/scripts/build-all-tools.sh

```diff
@@ -0,0 +1,105 @@
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-elixir:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/generic/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-generic:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/go/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-go:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/hcl/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-hcl:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/javascript/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-javascript:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/leaks/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-leaks:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/php/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-php:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/python/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-python:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
+    DOCKERFILE_PATH="$CURRENT_FOLDER/internal/services/formatters/shell/deployments/Dockerfile"
+    echo "Running Dockerfile: $DOCKERFILE_PATH ..."
+    if ! docker build -t "$IMAGE_BASE_NAME-shell:$IMAGE_TAG" -f "$DOCKERFILE_PATH" .; then
+        echo "Error on build docker file in path: $DOCKERFILE_PATH"
+        exit 1
+    fi
```

Comment:
This seems a lot of duplicated logic, what do you think about being more generic like this?
```bash
for DOCKERFILE in $(find internal -type f -iname "Dockerfile"); do
    LANGUAGE=$(echo $DOCKERFILE | cut -d "/" -f4)
    echo $LANGUAGE
    if ! docker build -t "$IMAGE_BASE_NAME-$LANGUAGE:$IMAGE_TAG" -f "$DOCKERFILE" .; then
        echo "Error on build docker file in path: $DOCKERFILE"
        exit 1
    fi
done
```

This logic will build any Dockerfile that exists on `internal` directory.

Reply:
i try this before using loop with `@ArrayString` but it not wait finish all builds to next step. I'll try this method

Reply:
i think mage could make your life easier on this case

Reply:
done @matheusalcantarazup